hedera-node/src/main/java/com/hedera/services/txns/consensus/TopicDeleteTransitionLogic.java

```diff
@@ -63,23 +56,10 @@ public void doStateTransition() {
 		var op = transactionContext.accessor().getTxn().getConsensusDeleteTopic();
 		var topicId = op.getTopicID();
 
-		var topicStatus = validator.queryableTopicStatus(topicId, topics.get());
-		if (OK != topicStatus) {
-			// Should not get here as the adminKey lookup should have failed.
-			transactionContext.setStatus(topicStatus);
-			return;
-		}
+		var topic = store.loadTopic(Id.fromGrpcTopic(topicId));
+		topic.setDeleted(true);
 
-		var topicMapKey = EntityNum.fromTopicId(topicId);
-		var topic = topics.get().get(topicMapKey);
-		if (!topic.hasAdminKey()) {
-			// Topics without adminKeys can't be deleted.
-			transactionContext.setStatus(UNAUTHORIZED);
-			return;
-		}
-
-		var mutableTopic = topics.get().getForModify(topicMapKey);
-		mutableTopic.setDeleted(true);
+		store.persistTopic(topic);
 
 		transactionContext.setStatus(SUCCESS);
```

Comment:
I think this should be done by the `TransitionRunner`. I think we missed adding  the `HederaFunctionality#TopicDelete` there.

Reply:
In the other PRs where the specialOps enum is defined, we will add it.